pkg/collector/corechecks/containers/containerd.go

```diff
@@ -0,0 +1,359 @@
+			log.Errorf("Could not process metadata related metrics for %s: %v", ctn.ID()[:12], err.Error())
+		}
+
+		if metrics.Memory.Size() > 0 {
+			computeMem(sender, metrics.Memory, tags)
+		}
+
+		if metrics.CPU.Throttling != nil && metrics.CPU.Usage != nil {
+			computeCPU(sender, metrics.CPU, tags)
+		}
+
+		if metrics.Blkio.Size() > 0 {
+			computeBlkio(sender, metrics.Blkio, tags)
+		}
+
+		if len(metrics.Hugetlb) > 0 {
+			computeHugetlb(sender, metrics.Hugetlb, tags)
+		}
+	}
+}
+
+func convertTasktoMetrics(task containerd.Task, nsCtx context.Context) (*cgroups.Metrics, error) {
+	metricTask, err := task.Metrics(nsCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	metricAny, err := typeurl.UnmarshalAny(&types.Any{
+		TypeUrl: metricTask.Data.TypeUrl,
+		Value:   metricTask.Data.Value,
+	})
+	if err != nil {
+		log.Errorf(err.Error())
+		return nil, err
+	}
+	return metricAny.(*cgroups.Metrics), nil
+}
+
+func computeExtra(sender aggregator.Sender, ctn containerd.Container, nsCtx context.Context, tags []string) error {
+	img, err := ctn.Image(nsCtx)
+	if err != nil {
+		return err
+	}
+	size, err := img.Size(nsCtx)
+	if err != nil {
+		return err
+	}
+	sender.Gauge("containerd.image.size", float64(size), "", tags)
+	return nil
+}
+
+func collectTags(ctn containerd.Container, nsCtx context.Context) ([]string, error) {
+	tags := []string{}
+
+	// Container image
+	im, err := ctn.Image(nsCtx)
+	if err != nil {
+		return tags, err
+	}
+	imageName := fmt.Sprintf("image:%s", im.Name())
+	tags = append(tags, imageName)
+
+	// Container labels
+	labels, err := ctn.Labels(nsCtx)
+	if err != nil {
+		return tags, err
+	}
+	for k, v := range labels {
+		tag := fmt.Sprintf("%s:%s", k, v)
+		tags = append(tags, tag)
+	}
+
+	// Container meta
+	i, err := ctn.Info(nsCtx)
+	if err != nil {
+		return tags, err
+	}
+
+	runt := fmt.Sprintf("runtime:%s", i.Runtime.Name)
+	tags = append(tags, runt)
+
+	return tags, nil
+}
+
+func computeHugetlb(sender aggregator.Sender, huge []*cgroups.HugetlbStat, tags []string) {
+	for _, h := range huge {
+		sender.Gauge("containerd.hugetlb.max", float64(h.Max), "", tags)
+		sender.Gauge("containerd.hugetlb.failcount", float64(h.Failcnt), "", tags)
+		sender.Gauge("containerd.hugetlb.usage", float64(h.Usage), "", tags)
+	}
+}
+
+func computeMem(sender aggregator.Sender, mem *cgroups.MemoryStat, tags []string) {
+	memList := map[string]*cgroups.MemoryEntry{
+		"containerd.mem.current":    mem.Usage,
+		"containerd.mem.kernel_tcp": mem.KernelTCP,
+		"containerd.mem.kernel":     mem.Kernel,
+		"containerd.mem.swap":       mem.Swap,
+	}
+	for metricName, memStat := range memList {
+		parseAndSubmitMem(metricName, sender, memStat, tags)
+	}
+	sender.Gauge("containerd.mem.cache", float64(mem.Cache), "", tags)
+	sender.Gauge("containerd.mem.rss", float64(mem.RSS), "", tags)
+	sender.Gauge("containerd.mem.rsshuge", float64(mem.RSSHuge), "", tags)
+	sender.Gauge("containerd.mem.usage", float64(mem.Usage.Usage), "", tags)
```

Comment:
What's the difference between:
  - `containerd.mem.usage` and `containerd.mem.current.usage` from parseAndSubmitMem
  - `containerd.mem.kernel.usage` and `containerd.mem.kernel.usage` from parseAndSubmitMem

Reply:
`containerd.mem.usage` and `containerd.mem.current.usage` the main difference is the word `current`... 😄 As for kernel ... 

More seriously - Good catch, removing